Sources/Sentry/SentryClient.m

```diff
@@ -472,13 +476,53 @@ - (void)captureSession:(SentrySession *)session
     }
 
     SentryEnvelopeItem *item = [[SentryEnvelopeItem alloc] initWithSession:session];
-    SentryEnvelopeHeader *envelopeHeader = [[SentryEnvelopeHeader alloc] initWithId:nil
-                                                                       traceContext:nil];
-    SentryEnvelope *envelope = [[SentryEnvelope alloc] initWithHeader:envelopeHeader
+    SentryEnvelope *envelope = [[SentryEnvelope alloc] initWithHeader:[SentryEnvelopeHeader empty]
                                                            singleItem:item];
     [self captureEnvelope:envelope];
 }
 
+- (void)captureReplayEvent:(SentryReplayEvent *)replayEvent
+           replayRecording:(SentryReplayRecording *)replayRecording
+                     video:(NSURL *)videoURL
+{
+    replayEvent = (SentryReplayEvent *)[self prepareEvent:replayEvent
+                                                withScope:[[SentryScope alloc] init]
+                                   alwaysAttachStacktrace:NO];
+
+    if (replayEvent == nil) {
+        return;
+    } else if (![replayEvent isKindOfClass:SentryReplayEvent.class]) {
+        SENTRY_LOG_DEBUG(@"The event preprocessor didn't update the replay event in place. The "
+                         @"replay was discarded.");
+        return;
+    }
+
+    // breadcrumbs for replay will be send with ReplayRecording
+    replayEvent.breadcrumbs = nil;
+
+    SentryEnvelopeItem *eventEnvelopeItem = [[SentryEnvelopeItem alloc] initWithEvent:replayEvent];
+
+    NSData *recording = [SentrySerialization dataWithJSONObject:[replayRecording serialize]];
+    SentryEnvelopeItem *recordingEnvelopeItem = [[SentryEnvelopeItem alloc]
+        initWithHeader:[SentryReplayEnvelopeItemHeader
+                           replayRecordingHeaderWithSegmentId:replayRecording.segmentId
+                                                       length:recording.length]
+                  data:recording];
+
+    NSData *video = [NSData dataWithContentsOfURL:videoURL];
```

Comment:
`m`: That part could be a bit heavy. Should we maybe put this into a background thread?

Furthermore, I think we need to fix this issue https://github.com/getsentry/sentry-cocoa/issues/3630 so we can use `NSFileHandle` or something similar to avoid copying the video into memory twice. We already get sometimes `NSMallocException`.

Reply:
We can make use of a background thread. 
The envelope format will change but I will look into avoiding load the video into memory.